rust/foxglove/src/sink.rs

```diff
@@ -1,13 +1,29 @@
 use crate::channel::Channel;
 use crate::metadata::Metadata;
 use crate::FoxgloveError;
+use std::sync::atomic::{AtomicU64, Ordering};
 use std::sync::Arc;
 
+/// Uniquely identifies a [`Sink`] in the context of this program.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct SinkId(u64);
+impl SinkId {
+    /// Allocates the next sink ID.
+    pub fn next() -> Self {
+        static NEXT_ID: AtomicU64 = AtomicU64::new(0);
```

Comment:
Channel IDs start at 1 for good reason. It doesn't really matter, but maybe it makes sense to be consistent here.